### ISSUE TITLE
The "Branch-Protection" check seems to fail to detect the status of "status checks"

### ISSUE THREAD

**Issue by evverx** [ossf/scorecard#1247]
I turned off the "Require status checks to pass before merging" setting and ran something like
```
curl  -u "***" -H "Accept: application/vnd.github.v3+json"   https://api.github.com/repos/evverx/systemd/branches/main/protection/required_status_checks
```
to make sure they are off:
```
{
  "message": "Required status checks not enabled",
  "documentation_url": "https://docs.github.com/rest/reference/repos#get-status-checks-protection"
}
```

`scorecard` says that "Info: strict status check enabled on branch main" and increments the score:
```
scorecard  --format json --show-details --verbosity Debug --checks Branch-Protection  --repo=https://github.com/evverx/systemd
...
"Info: strict status check enabled on branch 'main'","Warn: status checks for merging have no specific status to check on branch 'main'"
```

**Comment by laurentsimon** [comment]
Thanks for the report. This setting needs an admin token to be retrieved. We've prepared https://github.com/ossf/scorecard/pull/1243 to ignore the settings when it's not available.

**Comment by evverx** [comment]
I use the same token with both curl and scorecard so I'm not sure why the results would be different. Does `scorecard` uses some other endpoints to get the protection rules?

**Comment by laurentsimon** [comment]
we use graphQl APIs. @azeemsgoogle knows more about this part of the code

**Comment by evverx** [comment]
Fixed in #1252. Closing.
